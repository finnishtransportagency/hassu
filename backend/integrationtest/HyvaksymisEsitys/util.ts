import { config } from "../../src/config";
import { getS3Client } from "../../src/aws/client";
import { DeleteObjectsCommand, ListObjectsV2Command, ListObjectsV2CommandOutput, PutObjectCommand } from "@aws-sdk/client-s3";
import { createPresignedPost } from "@aws-sdk/s3-presigned-post";
import { adaptFileName } from "../../src/tiedostot/paths";
import { getAxios } from "../../src/aws/monitoring";
import FormData from "form-data";

export async function insertYllapitoFileToS3(pathInS3: string) {
  if (pathInS3.includes(config.yllapitoBucketName)) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  const buffer = Buffer.from("Test");
  await getS3Client().send(
    new PutObjectCommand({
      Body: buffer,
      Bucket: config.yllapitoBucketName,
      Key: pathInS3,
    })
  );
}

export async function insertUploadFileToS3(uuid: string, filename: string) {
  const fileNameWithPath = `${uuid}/${adaptFileName(filename)}`;
  const presignedPost = await createPresignedPost(getS3Client(), {
    Bucket: config.uploadBucketName,
    Key: fileNameWithPath,
    Expires: 600,
  });
  const fields = presignedPost.fields;
  const form = new FormData();
  Object.keys(fields).forEach((key) => {
    form.append(key, fields[key]);
  });
  form.append("file", "Test");
  const axios = getAxios();
  const lengthSync = form.getLengthSync();

  await axios.post(presignedPost.url, form, {
    headers: {
      "Content-Length": lengthSync,
      ...form.getHeaders(),
    },
  });
}

export async function deleteYllapitoFiles(sourcePrefix?: string) {
  if (sourcePrefix && sourcePrefix.includes(config.yllapitoBucketName)) {
    throw new Error("sourcePrefix ei saa sisältää yllapitoBucketNamea");
  }
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: sourcePrefix,
        ContinuationToken,
      })
    );
    await deleteFiles(Contents.map(({ Key }) => Key as string));

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

export async function emptyUploadFiles() {
  const s3 = getS3Client();
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.uploadBucketName,
        ContinuationToken,
      })
    );
    await deleteFiles(
      Contents.map(({ Key }) => Key as string),
      config.uploadBucketName
    );

    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
}

async function deleteFiles(paths: string[], bucket: string = config.yllapitoBucketName) {
  if (paths.some((path) => path.includes(config.yllapitoBucketName))) {
    throw new Error("path ei saa sisältää ylläpitoBucketNamea");
  }
  await Promise.all(
    // DeleteObjectsCommand takes at most 1000 items
    getChunksOfThousand(paths).map((paths) =>
      getS3Client().send(
        new DeleteObjectsCommand({
          Bucket: bucket,
          Delete: { Objects: paths.map((path) => ({ Key: path })) },
        })
      )
    )
  );
}

function getChunksOfThousand<T>(array: T[]): T[][] {
  const chunks: T[][] = [];
  for (let i = 0; i < array.length; i += 1000) {
    chunks.push(array.slice(i, i + 1000));
  }
  return chunks;
}

export async function getYllapitoFilesUnderPath(pathInS3: string): Promise<string[]> {
  const s3 = getS3Client();
  const files: string[] = [];
  let ContinuationToken = undefined;
  do {
    const { Contents = [], NextContinuationToken }: ListObjectsV2CommandOutput = await s3.send(
      new ListObjectsV2Command({
        Bucket: config.yllapitoBucketName,
        Prefix: pathInS3,
        ContinuationToken,
      })
    );
    files.push(...Contents.map(({ Key }) => Key as string));
    ContinuationToken = NextContinuationToken;
  } while (ContinuationToken);
  return files;
}
