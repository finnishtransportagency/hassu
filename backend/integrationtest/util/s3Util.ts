import { produceAWSClient } from "../../src/aws/clientProducer";
import { S3Client } from "@aws-sdk/client-s3";

export function localstackS3Client() {
  produceAWSClient(
    "s3",
    () =>
      new S3Client({
        endpoint: "http://localhost:4566",
        forcePathStyle: true,
      }),
    true
  );
}
