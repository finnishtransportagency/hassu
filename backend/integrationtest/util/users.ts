import * as userService from "../../src/service/userService";
import * as sinon from "sinon";
import { UserFixture } from "../../test/fixture/userFixture";

export function runAsVaylaUser() {
  sinon.stub(userService, "identifyUser").resolves();
  userService.identifyMockUser(UserFixture.mattiMeikalainen);
}

export function runAsAnonymous() {
  userService.identifyMockUser();
}
