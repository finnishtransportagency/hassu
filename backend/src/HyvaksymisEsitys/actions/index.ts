import tallennaHyvaksymisEsitys from "./tallenna";
import tallennaHyvaksymisEsitysJaLahetaHyvaksyttavaksi from "./tallennaJaLahetaHyvaksyttavaksi";
import avaaHyvaksymisEsityksenMuokkaus from "./avaaMuokkaus";
import hyvaksyHyvaksymisEsitys from "./hyvaksy";
import palautaHyvaksymisEsitys from "./palauta";
import suljeHyvaksymisEsityksenMuokkaus from "./suljeMuokkaus";
import listaaHyvaksymisEsityksenTiedostot from "./listaaTiedostot";
import { esikatseleHyvaksyttavaHyvaksymisEsityksenTiedostot, esikatseleHyvaksymisEsityksenTiedostot } from "./esikatseleTiedostot";
import haeHyvaksymisEsityksenTiedot from "./haeTiedot";

export {
  tallennaHyvaksymisEsitys,
  tallennaHyvaksymisEsitysJaLahetaHyvaksyttavaksi,
  avaaHyvaksymisEsityksenMuokkaus,
  hyvaksyHyvaksymisEsitys,
  palautaHyvaksymisEsitys,
  suljeHyvaksymisEsityksenMuokkaus,
  listaaHyvaksymisEsityksenTiedostot,
  esikatseleHyvaksymisEsityksenTiedostot,
  esikatseleHyvaksyttavaHyvaksymisEsityksenTiedostot,
  haeHyvaksymisEsityksenTiedot,
};
