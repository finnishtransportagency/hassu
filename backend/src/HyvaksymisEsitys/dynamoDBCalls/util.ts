import { UpdateCommand } from "@aws-sdk/lib-dynamodb";
import { log } from "../../logger";
import { getDynamoDBDocumentClient } from "../../aws/client";
import { ConditionalCheckFailedException } from "@aws-sdk/client-dynamodb";
import { SimultaneousUpdateError } from "hassu-common/error";

export async function sendUpdateCommandToDynamoDB(params: UpdateCommand): Promise<void> {
  if (log.isLevelEnabled("debug")) {
    log.debug("Updating projekti to Hassu with params", { params });
  }
  try {
    const dynamoDBDocumentClient = getDynamoDBDocumentClient();
    await dynamoDBDocumentClient.send(params);
  } catch (e) {
    if (e instanceof ConditionalCheckFailedException) {
      throw new SimultaneousUpdateError("Projektia on p√§ivitetty tietokannassa. Lataa projekti uudelleen.");
    }
    log.error(e instanceof Error ? e.message : String(e), { params });
    throw e;
  }
}
