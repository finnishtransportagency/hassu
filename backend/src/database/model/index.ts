export * from "./common";
export * from "./projekti";
export * from "./suunnitteluVaihe";
export * from "./nahtavillaoloVaihe";
export * from "./hyvaksymisPaatosVaihe";
