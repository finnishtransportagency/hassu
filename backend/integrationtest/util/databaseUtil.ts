import * as log from "loglevel";

import * as sinon from "sinon";
import { SinonStub } from "sinon";
import * as dynamoDB from "../../src/database/dynamoDB";

import DynamoDB from "aws-sdk/clients/dynamodb";
import { HookFunction } from "mocha";

const localDynamoDBParams = {
  endpoint: "http://localhost:4566",
  accessKeyId: "anyKey",
  secretAccessKey: "anySecret",
  region: "eu-west-1",
};
const localDynamoDB = new DynamoDB(localDynamoDBParams);
const localDocumentClient = new DynamoDB.DocumentClient({
  service: localDynamoDB,
  apiVersion: "2012-08-10",
  params: localDynamoDBParams,
});

// Try to use the database configuration from deployment in
let localDynamoDBDocumentClientStub: SinonStub;

function replaceAWSDynamoDBWithLocalstack() {
  localDynamoDBDocumentClientStub = sinon.stub(dynamoDB, "getDynamoDBDocumentClient");
  localDynamoDBDocumentClientStub.returns(localDocumentClient);
}

export async function setupLocalDatabase() {
  try {
    await deleteAllItemsFromDatabase();
  } catch (e) {
    // Ignore
  }
  replaceAWSDynamoDBWithLocalstack();
}

async function deleteAllItemsFromDatabase() {
  // Hard-code table name to prevent accidental deletion from AWS
  log.info("Cleaning up database");
  const items = (await localDocumentClient.scan({ TableName: "Projekti-localstack" }).promise()).Items;
  if (items) {
    await Promise.all(
      items.map(async (item) => {
        log.info("Deleting ", item);
        await localDocumentClient.delete({ TableName: "Projekti-localstack", Key: { oid: item.oid } }).promise();
      })
    );
  }
}

afterEach("Reset database stub", (() => {
  return localDynamoDBDocumentClientStub.restore();
}) as HookFunction);
