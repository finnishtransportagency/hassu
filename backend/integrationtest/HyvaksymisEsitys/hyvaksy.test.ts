import sinon from "sinon";
import * as API from "hassu-common/graphql/apiModel";
import { DBVaylaUser } from "../../src/database/model";
import { userService } from "../../src/user";
import TEST_HYVAKSYMISESITYS, {
  TEST_HYVAKSYMISESITYS2,
  TEST_HYVAKSYMISESITYS_FILES,
  TEST_HYVAKSYMISESITYS_FILES2,
} from "./TEST_HYVAKSYMISESITYS";
import {
  deleteYllapitoFiles,
  getProjektiFromDB,
  getYllapitoFilesUnderPath,
  insertProjektiToDB,
  insertYllapitoFileToS3,
  removeProjektiFromDB,
} from "./util";
import { hyvaksyHyvaksymisEsitys } from "../../src/HyvaksymisEsitys/actions";
import { omit } from "lodash";
import { expect } from "chai";
import { UserFixture } from "../../test/fixture/userFixture";
import { setupLocalDatabase } from "../util/databaseUtil";
import { IllegalAccessError, IllegalArgumentError } from "hassu-common/error";

describe("Hyväksymisesityksen hyväksyminen", () => {
  const userFixture = new UserFixture(userService);
  setupLocalDatabase();
  const oid = "Testi1";

  before(async () => {
    // Poista projektin tiedostot testisetin alussa
    await deleteYllapitoFiles(`yllapito/tiedostot/projekti/${oid}/`);
  });

  beforeEach(async () => {
    // Aseta muokattavalle hyväksymisesitykselle tiedostoja S3:een
    await Promise.all(
      TEST_HYVAKSYMISESITYS_FILES.map(async ({ path }) => {
        const fullpath = `yllapito/tiedostot/projekti/${oid}/muokattava_hyvaksymisesitys/${path}`;
        await insertYllapitoFileToS3(fullpath);
      })
    );
  });

  afterEach(async () => {
    // Poista projektin tiedostot joka testin päätteeksi
    await deleteYllapitoFiles(`yllapito/tiedostot/projekti/${oid}`);
    // Poista projekti joka testin päätteeksi
    await removeProjektiFromDB(oid);
    userFixture.logout();
  });

  after(() => {
    sinon.reset();
    sinon.restore();
  });

  it("päivittää muokattavan hyväksymisesityksen tilan ja palautusSyyn", async () => {
    userFixture.loginAsAdmin();
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.ODOTTAA_HYVAKSYNTAA, palautusSyy: "Virheitä" };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
    };
    await insertProjektiToDB(projektiBefore);
    await hyvaksyHyvaksymisEsitys({ oid, versio });
    const projektiAfter = await getProjektiFromDB(oid);
    expect({
      ...omit(projektiAfter, "paivitetty"),
      julkaistuHyvaksymisEsitys: omit(projektiAfter.julkaistuHyvaksymisEsitys, "hyvaksymisPaiva"),
    }).to.eql({
      ...projektiBefore,
      versio: versio + 1,
      muokattavaHyvaksymisEsitys: { ...muokattavaHyvaksymisEsitys, tila: API.HyvaksymisTila.HYVAKSYTTY, palautusSyy: null },
      julkaistuHyvaksymisEsitys: { ...omit(muokattavaHyvaksymisEsitys, ["tila", "palautusSyy"]), hyvaksyja: "theadminuid" },
    });
    expect(projektiAfter.paivitetty).to.exist;
    expect(projektiAfter.julkaistuHyvaksymisEsitys.hyvaksymisPaiva).to.exist;
  });

  it("onnistuu projektipäälliköltä ", async () => {
    const projari = UserFixture.pekkaProjari;
    const projariAsVaylaDBUser: Partial<DBVaylaUser> = {
      kayttajatunnus: projari.uid!,
      tyyppi: API.KayttajaTyyppi.PROJEKTIPAALLIKKO,
    };
    userFixture.loginAs(projari);
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.ODOTTAA_HYVAKSYNTAA, palautusSyy: "Virheitä" };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
      kayttoOikeudet: [projariAsVaylaDBUser],
    };
    await insertProjektiToDB(projektiBefore);
    const kutsu = hyvaksyHyvaksymisEsitys({ oid, versio });
    await expect(kutsu).to.be.eventually.fulfilled;
  });

  it("luo julkaistun hyväksymisesityksen muokattavan perusteella", async () => {
    const projari = UserFixture.pekkaProjari;
    const projariAsVaylaDBUser: Partial<DBVaylaUser> = {
      kayttajatunnus: projari.uid!,
      tyyppi: API.KayttajaTyyppi.PROJEKTIPAALLIKKO,
    };
    userFixture.loginAs(projari);
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.ODOTTAA_HYVAKSYNTAA, palautusSyy: "Virheitä" };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
      kayttoOikeudet: [projariAsVaylaDBUser],
    };
    await insertProjektiToDB(projektiBefore);
    await hyvaksyHyvaksymisEsitys({ oid, versio });
    const projektiAfter = await getProjektiFromDB(oid);
    expect(omit(projektiAfter.julkaistuHyvaksymisEsitys, "hyvaksymisPaiva")).to.eql({
      ...omit(muokattavaHyvaksymisEsitys, ["tila", "palautusSyy"]),
      hyvaksyja: projari.uid,
    });
    expect(projektiAfter.paivitetty).to.exist;
    expect(projektiAfter.julkaistuHyvaksymisEsitys.hyvaksymisPaiva).to.exist;
  });

  it("poistaa vanhat julkaistut tiedostot ja kopioi muokkaustilaisen hyväksymisesityksen tiedostot julkaisulle", async () => {
    const projari = UserFixture.pekkaProjari;
    const projariAsVaylaDBUser: Partial<DBVaylaUser> = {
      kayttajatunnus: projari.uid!,
      tyyppi: API.KayttajaTyyppi.PROJEKTIPAALLIKKO,
    };
    userFixture.loginAs(projari);
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.ODOTTAA_HYVAKSYNTAA, palautusSyy: "Virheitä" };
    const julkaistuHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS2, hyvaksyja: projari.uid, hyvaksymisPaiva: "2022-01-01" };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
      julkaistuHyvaksymisEsitys,
      kayttoOikeudet: [projariAsVaylaDBUser],
    };
    await insertProjektiToDB(projektiBefore);
    // Aseta julkaistulle hyväksymisesitykselle tiedostoja S3:een
    await Promise.all(
      TEST_HYVAKSYMISESITYS_FILES2.map(async ({ path }) => {
        const fullpath = `yllapito/tiedostot/projekti/${oid}/hyvaksymisesitys/${path}`;
        await insertYllapitoFileToS3(fullpath);
      })
    );
    await hyvaksyHyvaksymisEsitys({ oid, versio });
    const files = await getYllapitoFilesUnderPath(`yllapito/tiedostot/projekti/${oid}`);
    expect(files).to.eql([
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/hyvaksymisEsitys/hyvaksymisEsitys_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/kuulutuksetJaKutsu/kuulutuksetJaKutsu_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/lausunnot/lausunnot_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/maanomistajaluettelo/maanomistajaluettelo_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/muistutukset/muistutukset_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/muuAineistoKoneelta/muuAineistoKoneelta_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/muuAineistoVelhosta/muuAineistoVelhosta_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/hyvaksymisesitys/suunnitelma/suunnitelma_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/hyvaksymisEsitys/hyvaksymisEsitys_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/kuulutuksetJaKutsu/kuulutuksetJaKutsu_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/lausunnot/lausunnot_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/maanomistajaluettelo/maanomistajaluettelo_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/muistutukset/muistutukset_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/muuAineistoKoneelta/muuAineistoKoneelta_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/muuAineistoVelhosta/muuAineistoVelhosta_aoa_.png",
      "yllapito/tiedostot/projekti/Testi1/muokattava_hyvaksymisesitys/suunnitelma/suunnitelma_aoa_.png",
    ]);
  });

  it("ei onnistu projektihenkilöltä", async () => {
    const projari = UserFixture.pekkaProjari;
    const projariAsVaylaDBUser: Partial<DBVaylaUser> = {
      kayttajatunnus: projari.uid!,
      tyyppi: API.KayttajaTyyppi.PROJEKTIPAALLIKKO,
    };
    const muokkaaja = UserFixture.manuMuokkaaja;
    const muokkaajaAsVaylaDBUser: Partial<DBVaylaUser> = {
      kayttajatunnus: muokkaaja.uid!,
    };
    userFixture.loginAs(muokkaaja);
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.ODOTTAA_HYVAKSYNTAA, palautusSyy: "Virheitä" };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
      kayttoOikeudet: [projariAsVaylaDBUser, muokkaajaAsVaylaDBUser],
    };
    await insertProjektiToDB(projektiBefore);
    const kutsu = hyvaksyHyvaksymisEsitys({ oid, versio });
    await expect(kutsu).to.be.eventually.be.rejectedWith(IllegalAccessError);
  });

  it("ei onnistu, jos muokattava hyväksymiseistys on hyväksytty-tilassa", async () => {
    userFixture.loginAsAdmin();
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.HYVAKSYTTY };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
    };
    await insertProjektiToDB(projektiBefore);
    const kutsu = hyvaksyHyvaksymisEsitys({ oid, versio });
    await expect(kutsu).to.be.eventually.be.rejectedWith(IllegalArgumentError);
  });

  it("ei onnistu, jos muokattava hyväksymiseistys on muokkaustilassa", async () => {
    userFixture.loginAsAdmin();
    const muokattavaHyvaksymisEsitys = { ...TEST_HYVAKSYMISESITYS, tila: API.HyvaksymisTila.MUOKKAUS };
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
      muokattavaHyvaksymisEsitys,
    };
    await insertProjektiToDB(projektiBefore);
    const kutsu = hyvaksyHyvaksymisEsitys({ oid, versio });
    await expect(kutsu).to.be.eventually.be.rejectedWith(IllegalArgumentError);
  });

  it("ei onnistu, jos muokattavaa hyväksymisesitystä ei ole", async () => {
    userFixture.loginAsAdmin();
    const versio = 2;
    const projektiBefore = {
      oid,
      versio,
    };
    await insertProjektiToDB(projektiBefore);
    const kutsu = hyvaksyHyvaksymisEsitys({ oid, versio });
    await expect(kutsu).to.be.eventually.be.rejectedWith(IllegalArgumentError);
  });
});
