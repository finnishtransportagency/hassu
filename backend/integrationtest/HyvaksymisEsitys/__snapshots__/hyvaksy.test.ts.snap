// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun EI tarvita kiireellistä käsittelyä ja asianhallinta on aktiivinen 1`] = `
Object {
  "subject": "Valtion liikenneväylien suunnittelu: hyväksymisesitys hyväksytty asiatunnusELY",
  "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi Projektin nimi hyväksymisesitys on hyväksytty.

Voit tarkastella hyväksymisesitystä osoitteessa https://localhost:3000/yllapito/projekti/Testi1/hyvaksymisesitys

Järjestelmä vie automaattisesti tarpeelliset tiedostot asianhallintaan. Käythän kuitenkin tarkistamassa asianhallinnan getParameterValue_UspaBaseUrl/Asia.aspx?AsiaId=14 .

Sait tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
  "to": Array [
    "email@email.com",
    "email2@email.com",
  ],
}
`;

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun EI tarvita kiireellistä käsittelyä ja asianhallinta on aktiivinen 2`] = `
Object {
  "subject": "Valtion liikenneväylien suunnittelu: hyväksymisesitys hyväksytty asiatunnusELY",
  "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi Projektin nimi hyväksymisesitys on hyväksytty.

Voit tarkastella hyväksymisesitystä osoitteessa https://localhost:3000/yllapito/projekti/Testi1/hyvaksymisesitys
    
Sait tämän viestin, koska sinut on merkitty hyväksymisesityksen laatijaksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
  "to": "muokkaaja@email.com",
}
`;

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun EI tarvita kiireellistä käsittelyä ja asianhallinta on aktiivinen 3`] = `
Object {
  "cc": Array [
    "email@email.com",
    "email2@email.com",
  ],
  "subject": "Kiire hyväksymisesitys Projektin nimi",
  "text": "Kiire hyväksymisesitys Projektin nimi

ELY-keskus lähettää suunnitelman Projektin nimi hyväksyttäväksi Traficomiin kiireellisenä. Suunnitelman hyväksymisesitys ja laskutustiedot hyväksymismaksua varten löytyy oheisen linkin takaa https://localhost:3000/suunnitelma/Testi1/hyvaksymisesityksen-aineistot

Sähköpostin liitteenä on myös hyväksymisesitys.

Lisätiedot 

Lisätietoja



Laskutustiedot hyväksymismaksua varten

Suunnitelman nimi

Projektin nimi

Asiatunnus

asiatunnusELY

Vastuuorganisaatio

Lapin ELY-keskus

Y-tunnus

2296962-1

OVT-tunnus

ovtTunnus

Verkkolaskuoperaattorin välittäjätunnus

verkkolaskuoperaattorinTunnus

Viite

viitetieto



Lisätietoja suunnitelmasta antaa 

Etunimi Sukunimi Hämeen ELY-keskus, 

puh 0291234567,

email@email.com


Tämä viesti on lähetetty automaattisesti Valtion liikenneväylien suunnittelu -järjestelmän kautta eikä siihen voi vastata.",
  "to": Array [
    "vastaanottaja@sahkoposti.fi",
  ],
}
`;

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun tarvitaan kiireellistä käsittelyä ja asianhallinta ei ole aktiivinen 1`] = `
Object {
  "subject": "Valtion liikenneväylien suunnittelu: hyväksymisesitys hyväksytty asiatunnusELY",
  "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi Projektin nimi hyväksymisesitys on hyväksytty.

Voit tarkastella hyväksymisesitystä osoitteessa https://localhost:3000/yllapito/projekti/Testi1/hyvaksymisesitys

Viethän hyväksymisesityksen asianhallintaan suunnitelman hallinnollisen käsittelyn asialle. Toimi organisaatiosi asianhallinnan ohjeistusten mukaisesti.

Sait tämän viestin, koska sinut on merkitty projektin projektipäälliköksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
  "to": Array [
    "email@email.com",
    "email2@email.com",
  ],
}
`;

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun tarvitaan kiireellistä käsittelyä ja asianhallinta ei ole aktiivinen 2`] = `
Object {
  "subject": "Valtion liikenneväylien suunnittelu: hyväksymisesitys hyväksytty asiatunnusELY",
  "text": "Valtion liikenneväylien suunnittelu -järjestelmän projektisi Projektin nimi hyväksymisesitys on hyväksytty.

Voit tarkastella hyväksymisesitystä osoitteessa https://localhost:3000/yllapito/projekti/Testi1/hyvaksymisesitys
    
Sait tämän viestin, koska sinut on merkitty hyväksymisesityksen laatijaksi. Tämä on automaattinen sähköposti, johon ei voi vastata.",
  "to": "muokkaaja@email.com",
}
`;

exports[`Hyväksymisesityksen hyväksyminen lähettää oikeat s.postit kun tarvitaan kiireellistä käsittelyä ja asianhallinta ei ole aktiivinen 3`] = `
Object {
  "cc": Array [
    "email@email.com",
    "email2@email.com",
  ],
  "subject": "Kiire hyväksymisesitys Projektin nimi",
  "text": "Kiire hyväksymisesitys Projektin nimi

ELY-keskus lähettää suunnitelman Projektin nimi hyväksyttäväksi Traficomiin kiireellisenä. Suunnitelman hyväksymisesitys ja laskutustiedot hyväksymismaksua varten löytyy oheisen linkin takaa https://localhost:3000/suunnitelma/Testi1/hyvaksymisesityksen-aineistot

Sähköpostin liitteenä on myös hyväksymisesitys.

Lisätiedot 

Lisätietoja



Laskutustiedot hyväksymismaksua varten

Suunnitelman nimi

Projektin nimi

Asiatunnus

asiatunnusELY

Vastuuorganisaatio

Lapin ELY-keskus

Y-tunnus

2296962-1

OVT-tunnus

ovtTunnus

Verkkolaskuoperaattorin välittäjätunnus

verkkolaskuoperaattorinTunnus

Viite

viitetieto



Lisätietoja suunnitelmasta antaa 

Etunimi Sukunimi Hämeen ELY-keskus, 

puh 0291234567,

email@email.com


Tämä viesti on lähetetty automaattisesti Valtion liikenneväylien suunnittelu -järjestelmän kautta eikä siihen voi vastata.",
  "to": Array [
    "vastaanottaja@sahkoposti.fi",
  ],
}
`;
